src/lib/logical_query_plan/jit_aware_lqp_translator.cpp

```diff
@@ -309,14 +309,15 @@ bool is_between_jit_expression_type_upper_inclusive(JitExpressionType jit_expres
          jit_expression_type == JitExpressionType::BetweenLowerExclusive;
 }
 
-std::shared_ptr<const JitExpression> JitAwareLQPTranslator::_try_translate_expression_to_jit_expression(
+std::shared_ptr<JitExpression> JitAwareLQPTranslator::_try_translate_expression_to_jit_expression(
     const std::shared_ptr<AbstractExpression>& expression, JitReadTuples& jit_source,
     const std::shared_ptr<AbstractLQPNode>& input_node, const bool use_actual_value) const {
   const auto input_node_column_id = input_node->find_column_id(*expression);
   if (input_node_column_id) {
-    const auto tuple_entry = jit_source.add_input_column(
-        expression->data_type(), input_node->is_column_nullable(input_node->get_column_id(*expression)),
-        *input_node_column_id, use_actual_value);
+    // The correct nullable information is set when the JitOperatorWrapper is executed with access to the input table.
+    const bool is_nullabe = true;
```

Comment:
typo. Also, it leaves us with wrong information for a while.`optional` or `bool guaranteed_non_null`?

Reply:
If you don’t want an optional, having a flag "guaranteed to not be nullable” is different from “is_nullable” (or its opposite). For the former, `false` means “might or might not be nullable”, while it means “definitely not nullable” for the latter.

This would just be a naming change but would somewhat (but not completely) eliminate the problem with the incomplete state.